cloud/amazon/route53.py

```diff
@@ -391,7 +391,10 @@ def main():
         #Need to save this changes in rset, because of comparing rset.to_xml() == wanted_rset.to_xml() in next block
 	rset.name = decoded_name
 
-        if rset.type == type_in and decoded_name.lower() == record_in.lower() and str(rset.identifier) == str(identifier_in):
+        if identifier_in is not None:
+            identifier_in = str(identifier_in)
+	
+        if rset.type == type_in and decoded_name.lower() == record_in.lower() and rset.identifier == identifier_in:
```

Comment:
Curious why str(rset.identifier) was dropped? I see what you did with identifier_in, and that makes sense.